src/chat/__tests__/chatReducers-test.js

```diff
@@ -656,5 +656,25 @@ describe('chatReducers', () => {
 
       expect(newState).toEqual(expectedState);
     });
+
+    test('when replaceExisting is true, common messages are not replaced', () => {
```

Comment:
I missed this... updated to perform a deep equal.

Now one thing to consider is if the win we get from not re-rendering better than the cost for performing a deep equal. For that I have timed the startup for both cases, and gonna continue this discuss though chat.